Fix #18193 don't trigger file deprecation transitively (#18349)
coq/coq: doc/sphinx/using/libraries/writing.rst

             Ltac2 definition @qualid is deprecated since @string__since. @string__note
             Ltac2 alias @qualid is deprecated since @string__since. @string__note
             Ltac2 notation {+ @ltac2_scope } is deprecated since @string__since. @string__note
             Library File @qualid is deprecated since @string__since. @string__note

>              Library File (transitively required) @qualid is deprecated since @string__since. @string__note

Reviewer: Maybe this would be better in alphabetic order?  The rationale behind the current order is unclear.

      :n:`@qualid` or :n:`@string` is the notation,
      :n:`@string__since` is the version number, :n:`@string__note` is
      the note (usually explains the replacement).

>       Explicitly :cmd:`Require` ing a file that has been deprecated,

Reviewer: ```suggestion
      Explicitly :cmd:`Require`\ing a file that has been deprecated,
```

I forgot the idiom to omit the space before.

      using the :cmd:`Attributes` command, triggers a ``Library File``
      deprecation warning. Requiring a deprecated file, even indirectly through a chain of

>       :cmd:`Require` s, will produce a

Reviewer: ```suggestion
      :cmd:`Require`\s, will produce a
```

>       deprecation warning. Any loading (even indirectly, through a chain of
>       requirements) of such a file triggers the
>       ``Library File (transitively required)`` deprecation warning,
>       which is silenced by default. The latter can be unsilenced with
>       the :opt:`Warnings` option "deprecated-transitive-library-file".

Reviewer: ```suggestion
      deprecation warning.  Loading a deprecated file indirectly through a chain of
      :cmd:`Require`s will produce a 
      ``Library File (transitively required)`` deprecation warning if
       the :opt:`Warnings` option "deprecated-transitive-library-file" is enabled.
      (It is disabled by default.)
```

Saying "Loading" suggests using the "Load" command, if that's not correct, maybe say ```:cmd:`Require`ing```?

I'm guessing that you don't get the transitive message for something that's directly loaded.

Are the names of warnings in  `Attributes warning = ...`, `Set Warning = ...` and `[# warnings = "..."]` identical?  I don't see that mentioned in the doc though I didn't do a complete search.  If correct, it would be great and if you're could add a sentence to the description of each of the three constructs!  But a little more than what you planned to do.

Author: > Saying "Loading" suggests using the "Load" command, if that's not correct, maybe say :cmd:`Require`ing?

Indeed, thanks.

> I'm guessing that you don't get the transitive message for something that's directly loaded.

No, you get it in any case (easier to implement and likely more convenient to use), updated the sentence accordingly.

> Are the names of warnings in Attributes warning = ..., Set Warning = ... and [# warnings = "..."] identical?

The first one doesn't exist and I don't even know what it would mean (the `Attributes` command only handles the `deprecated` attribute currently and errors on any other attribute). The latter two are identical as documented in ``:opt:`Warnings` `` that itself links to the `#[warnings=` attribute.
